### PR TITLE
Check that a remote update exists even if a pending update is cached

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:

None

### Summary of the issue:

The following was reported in a user group. When running "checkForUpdate" from beta5, someone encountered the following bug and log.

```python
IO - speech.speech.speak (11:22:14.827) - MainThread (3288):
Speaking [LangChangeCommand ('en_GB'), 'Checking for Update', 'dialog', 'Checking for update', CancellableSpeech (still valid)]
IO - tones.beep (11:22:15.231) - MainThread (3288):
Beep at pitch 1760, for 40 ms, left volume 50, right volume 50
DEBUG - gui.contextHelp.bindHelpEvent (11:22:15.232) - MainThread (3288):
Did context help binding for UpdateResultDialog
ERROR - unhandled exception (11:22:15.232) - MainThread (3288):
Traceback (most recent call last):
 File "wx\core.pyc", line 3407, in <lambda>
 File "updateCheck.pyc", line 343, in __init__
TypeError: 'NoneType' object is not subscriptable
```

It seems that if a pending update is cached, even if there is no remote update, we attempt to compare the versions. This seems like an uncommon bug, possibly brought to surface by the earlier beta updates failing.

### Description of how this pull request fixes the issue:

Added a trail of typing to confirm the possibility of a state where there is a pending update cached and no remote update. Added a check to ensure that both of these exist before comparing their versions.

### Testing strategy:

As with any updating issue, this must be confirmed on alpha. 

### Known issues with pull request:

I am not certain that this is the cause of the issue, and without more information from the user it is hard to be certain as I was unable to reproduce this without spoofing a cached update.

### Change log entries:

None needed as it seems to be a rare, new and minimally obtrusive bug.

### Code Review Checklist:

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [ ] System (end to end) tests.
- [ ] Manual testing.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers

### CODE REVIEW

File: source/updateCheck.py [hunk 1]
```diff
@@ -331,20 +331,22 @@ class UpdateResultDialog(
...
 		super().__init__(parent, title=_("NVDA Update"))
 
 		self.updateInfo = updateInfo
 		mainSizer = wx.BoxSizer(wx.VERTICAL)
 		sHelper = guiHelper.BoxSizerHelper(self, orientation=wx.VERTICAL)
 
+		remoteUpdateExists = updateInfo is not None
 		pendingUpdateDetails = getPendingUpdate()
-		canOfferPendingUpdate = isPendingUpdate() and pendingUpdateDetails[1] == updateInfo["version"]
+		remoteUpdateMatchesPending = pendingUpdateDetails[1] == updateInfo["version"]
```
Comment (michaelDCurran) [inline source/updateCheck.py:344]: Shouldn't this line check remoteUpdateExists as well? Otherwise is rupdateInfo is None there will be a KeyError.

**Review by michaelDCurran — CHANGES_REQUESTED** [review]

File: source/updateCheck.py (hunk 1 above)
Comment (seanbudd) [reply source/updateCheck.py]: oops, my linting fix reintroduced the problem

**Review by michaelDCurran — APPROVED** [review]

**Comment by Brian1Gaff** [comment]
At the moment  the suggested input in the console fixed the issue on b5, and 
yes I guess there could have been a pending update from one of the non 
working versions prior to this one as it had to be installed manually, which 
of course did not check for any cached updates as it was a forced install.
 In my copy of alpha I have no issues, but I cannot honestly recall if there 
was a cached update when I over wrote that one. I suspect not, since the 
instructions were to perform the manual update there as well.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

**Comment by Brian1Gaff** [comment]
Maybe it should just delete any pending updates if you are installing 
manually and not from an internal update, IE then when you did the check on 
updates on the manually installed version there would be no impending 
updates or flags in files present and you would have a clean start?
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

**Pull request merged**